docs/CODING_STYLE.md

```diff
@@ -84,7 +86,25 @@ layout: default
 
 - Do not write functions that clobber call-by-reference variables on
   failure. Use temporary variables for these cases and change the passed in
-  variables only on success.
+  variables only on success. The rule is: never clobber return parameters on
+  failure, always initialize return parameters on success.
+
+- Typically, function parameters fit into three categories: input parameters,
+  mutable objects, and call-by-reference return parameters. Input parameters
```

Comment:
call-by-reference? No idea what that is.

Reply:
I assumed it to be a fairly well understood technical term:

https://en.wikipedia.org/wiki/Evaluation_strategy#Call_by_reference

Just means instead of passing a value, you pass a reference (in C: a pointer) to a value. for input params we typically do both in C: call-by-value (for simple types) and call-by-ref (for complex types, such as strings, structs, arrays). For return params call-by-value doesn't work though, hence we always do call-by-ref in C.

I think we should leave this in, I still believe it's a fairly well understood concept, and wikipedia will point people to the explanation quickly if they just type it in.

Reply:
I always heard that as "pass by reference". It probably varies by subculture or something.

Reply:
actually, admittedly, "pass-by-ref" makes more sense than "call-by-ref", but i learnt it by that and wikipedia agrees with my teachers ;-)